### DISCUSSION TITLE
How do you handle loading more content on your statically generated sites?

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/next.js#69917] [Help]
### Summary

For a lot of the sites I build, statically generating a single page is quite easy. Next.js provides `generateStaticParams` which we can use to tell Next.js what pages to statically render.

What I always have to make by scratch however is pages that list other pages. On an index page for example, I can't simply statically render a link to every single page on the website (I could, but Google probably wouldn't be happy), so I find myself generating JSON files inside my `/public` folder that I can then fetch on the client-side.

For example, if I want to generate paginated JSON of my content I add the following to my `generateStaticParams` function:

```
export const generateStaticParams = async () => {
  const blog = await getAllMdx<Blog>("blog");
  if (!fs.existsSync(process.cwd() + "/public/blog")) {
    await fs.promises.mkdir(process.cwd() + "/public/blog");
  }

  // divide all content into pages
  const chunkedBlog = chunkArray(blog);

  for (const [index, c] of chunkedBlog.entries()) {
    await fs.promises.writeFile(
      process.cwd() + `/public/blog/${index + 1}.json`,
      JSON.stringify(b)
    );
  }

  return [];
};
```

Then when I need to fetch a second page I can do
```
fetch(`/blog/2.json`)
```

It works, but it's not _great_.

How do you handle this on your side?

**Comment by Shriharsh-Deshmukh** [comment]
Hey  **@leo-petrucci**! When it comes to loading more content on statically generated sites, you're on the right track by generating JSON for paginated content. Here's a friendly suggestion on how to improve it:

Instead of manually creating JSON files and fetching them, you can leverage **Incremental Static Regeneration (ISR)** in Next.js. With ISR, you can generate static pages on-demand, which helps with performance and SEO. You don't have to pre-generate all pages—just the initial ones, and Next.js will handle the rest when users request them.

Here’s a quick example:
```js
export async function getStaticProps() {
  const blog = await getAllMdx<Blog>("blog");

  return {
    props: {
      blog,
    },
    revalidate: 60, // ISR - re-generates every 60 seconds
  };
}
```
This approach reduces the need for manual JSON management and improves scalability. You can still paginate but without the overhead of fetching external JSON files.

Hope this helps!
